api/admin/controller.py

```diff
@@ -321,40 +321,51 @@ def edit(self, data_source, identifier_type, identifier):
             work.calculate_presentation(policy=policy)
         return Response("", 200)
 
-    def suppress(self, data_source, identifier_type, identifier):
+    def suppress(self, identifier_type, identifier):
         """Suppress the license pool associated with a book."""
-        
         # Turn source + identifier into a LicensePool
-        pool = self.load_licensepool(data_source, identifier_type, identifier)
-        if isinstance(pool, ProblemDetail):
+        pools = self.load_licensepools(self.library, identifier_type, identifier)
+        if isinstance(pools, ProblemDetail):
             # Something went wrong.
-            return pool
-    
-        pool.suppressed = True
+            return pools
+
+        # Assume that the Work is being suppressed from the catalog, and
+        # not just the LicensePool.
+        # TODO: Suppress individual LicensePools when it's not that deep.
+        for pool in pools:
+            pool.suppressed = True
         return Response("", 200)
 
-    def unsuppress(self, data_source, identifier_type, identifier):
+    def unsuppress(self, identifier_type, identifier):
         """Unsuppress the license pool associated with a book."""
-        
         # Turn source + identifier into a LicensePool
-        pool = self.load_licensepool(data_source, identifier_type, identifier)
-        if isinstance(pool, ProblemDetail):
+        pools = self.load_licensepools(self.library, identifier_type, identifier)
+        if isinstance(pools, ProblemDetail):
             # Something went wrong.
-            return pool
-    
-        pool.suppressed = False
+            return pools
+
+        for pool in pools:
+            # Only unsuppress LicensePools that with Work-targetted,
+            # admin-editable complaints. Anything specific to a
+            # LicensePool is ignored until its specific complaints have
+            # been resolved.
+            ignore_pool = any(c.for_license_pool for c in pool.complaints)
+            if not ignore_pool:
+                pool.suppressed = False
```

Comment:
I have some concerns about this change. Unsuppressing is no longer the reverse of unsuppressing, so if an admin suppressed something accidentally it would be harder to undo. 

It might be a good idea to require admins to mark outstanding complaints resolved before unsuppressing, but if you try to resolve when a pool still has complaints you should get a problem detail. And I don't think it should be limited to license-pool-specific complaints, and we should also require the admin to add a complaint before suppressing a pool that doesn't have one yet. 

I think for now we should unsuppress all the pools regardless of whether there are complaints, and revisit this later.